src/Microsoft.Dnx.ApplicationHost/Program.cs

```diff
@@ -36,47 +36,56 @@ public Task<int> Main(string[] args)
+                if (options.WatchFiles)
+                {
+                    watcher = new FileWatcher(ProjectRootResolver.ResolveRootDirectory(Path.GetFullPath(options.ApplicationBaseDirectory)));
+                }
+                else
+                {
+                    watcher = NoopWatcher.Instance;
+                }
 
-            var host = new DefaultHost(options, _serviceProvider, _loadContextAccessor, watcher);
+                host = new DefaultHost(options, _serviceProvider, _loadContextAccessor, watcher);
 
-            if (host.Project == null)
-            {
-                return Task.FromResult(-1);
-            }
+                if (host.Project == null)
+                {
+                    return Task.FromResult(-1);
+                }
 
-            var lookupCommand = string.IsNullOrEmpty(options.ApplicationName) ? "run" : options.ApplicationName;
-            string replacementCommand;
-            if (host.Project.Commands.TryGetValue(lookupCommand, out replacementCommand))
-            {
-                // preserveSurroundingQuotes: false to imitate a shell. Shells remove quotation marks before calling
-                // Main methods. Here however we are invoking Main() without involving a shell.
-                var replacementArgs = CommandGrammar
-                    .Process(replacementCommand, GetVariable, preserveSurroundingQuotes: false)
-                    .ToArray();
-                options.ApplicationName = replacementArgs.First();
-                programArgs = replacementArgs.Skip(1).Concat(programArgs).ToArray();
-            }
+                var lookupCommand = string.IsNullOrEmpty(options.ApplicationName) ? "run" : options.ApplicationName;
+                string replacementCommand;
+                if (host.Project.Commands.TryGetValue(lookupCommand, out replacementCommand))
+                {
+                    // preserveSurroundingQuotes: false to imitate a shell. Shells remove quotation marks before calling
+                    // Main methods. Here however we are invoking Main() without involving a shell.
+                    var replacementArgs = CommandGrammar
+                        .Process(replacementCommand, GetVariable, preserveSurroundingQuotes: false)
+                        .ToArray();
+                    options.ApplicationName = replacementArgs.First();
+                    programArgs = replacementArgs.Skip(1).Concat(programArgs).ToArray();
+                }
 
-            if (string.IsNullOrEmpty(options.ApplicationName) ||
-                string.Equals(options.ApplicationName, "run", StringComparison.Ordinal))
+                if (string.IsNullOrEmpty(options.ApplicationName) ||
+                    string.Equals(options.ApplicationName, "run", StringComparison.Ordinal))
+                {
+                    options.ApplicationName = host.Project.EntryPoint ?? host.Project.Name;
+                }
+            }
+            catch (Exception ex)
```

Comment:
Going to change this since @davidfowl didn't like this. We'll throw here and catch/handle exception suppression in `RuntimeBootstrapper.cs` instead.